subprojects/tooling-api/src/integTest/groovy/org/gradle/integtests/tooling/r31/PersistentCompositeDependencySubstitutionCrossVersionSpec.groovy

```diff
@@ -102,7 +104,10 @@ class PersistentCompositeDependencySubstitutionCrossVersionSpec extends ToolingA
         ideaModule.dependencies.size() == 1
         with(ideaModule.dependencies.first()) {
             it instanceof IdeaModuleDependency
-            dependencyModule == null
+            // this currently fails for current -> 3.5
+            if (targetVersion < GradleVersion.version("4.0")) {
```

Comment:
Is it troublesome that we don't do this differentiation elsewhere?  Is this simply the first time we've removed methods from the Tooling API?